core/src/test/java/com/novoda/merlin/service/PingShould.java

```diff
@@ -1,71 +1,104 @@
 package com.novoda.merlin.service;
 
+import com.novoda.merlin.service.request.RequestException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.experimental.runners.Enclosed;
 import org.junit.runner.RunWith;
-import org.junit.runners.JUnit4;
+import org.junit.runners.Parameterized;
 import org.mockito.Mock;
 
 import static org.fest.assertions.api.Assertions.assertThat;
 import static org.mockito.Mockito.when;
 import static org.mockito.MockitoAnnotations.initMocks;
 
-@RunWith(JUnit4.class)
+@RunWith(Enclosed.class)
 public class PingShould {
 
-    private static final int OK = 200;
-    private static final int CREATED = 201;
-    private static final int NO_CONTENT = 204;
-    private static final int MOVED_PERMANENTLY = 301;
     private static final String HOST_ADDRESS = "any host address";
 
-    @Mock
-    private HostPinger.PingerCallback mockPingerCallback;
+    @RunWith(Parameterized.class)
+    public static class GivenSuccessfulRequest {
 
-    @Mock
-    private HostPinger.ResponseCodeFetcher mockResponseCodeFetcher;
+        @Parameterized.Parameters(name = "{0}")
+        public static Collection<Integer[]> data() {
+            return ResponseCode.toParameterList();
+        }
 
-    private Ping ping;
+        private final Ping ping;
 
-    @Before
-    public void setUp() throws Exception {
-        initMocks(this);
-        ping = new Ping(HOST_ADDRESS, mockResponseCodeFetcher);
-    }
+        @Mock
+        private HostPinger.ResponseCodeFetcher mockResponseCodeFetcher;
 
-    @Test
-    public void returnsTrueWhenHostResponseCodeIsSuccess() throws Exception {
-        when(mockResponseCodeFetcher.from(HOST_ADDRESS)).thenReturn(OK);
+        private final int responseCode;
 
-        boolean isSuccess = ping.doSynchronousPing();
+        public GivenSuccessfulRequest(int responseCode) {
+            initMocks(this);
```

Comment:
not using `@Rule` because the version of Mockito we're using doesn't have it yet. Maybe worth opening an issue?